here-naksha-app-service/src/main/java/com/here/naksha/app/service/http/tasks/AbstractApiTask.java

```diff
@@ -187,6 +195,17 @@ private XyzResponse handleNoElements(NoElementsStrategy noElementsStrategy) {
         if (wrResult instanceof ContextXyzFeatureResult cr) {
           violations = cr.getViolations();
         }
+        if (Objects.equals(type, Storage.class)) {
+          for (R feature : insertedFeatures) {
+            removePasswordFromFeature(feature);
+          }
+          for (R feature : updatedFeatures) {
+            removePasswordFromFeature(feature);
+          }
+          for (R feature : deletedFeatures) {
+            removePasswordFromFeature(feature);
+          }
+        }
```

Comment:
One thing I noticed is that new logic applies only to `Storage` class and therefore I think it would be better not to put in `AbstractApiTask`, and rather in `StorageApiTask` - it's not a behavior to be inherited for every API of ours.
This would mean little refactoring as we would need to somehow hijack `transformWriteResultToXyzCollectionResponse` (we need to modify response before sending it via verticle). This made me think that maybe we should split `transformWriteResultToXyzCollectionResponse` method which currently does two things
1) convert result to responseBody (ie `XyzFeatureCollection`)
2) send the responseBody via `sendXyzResponse`

If we split these in two, then clients of this API would have a chance to modify the outcome of operation 1 (parsing) before excuting operation 2 (sending response).

Let me know what you think about it :)

Reply:
per our private message this would take me a bit longer, so I made a subtask for it MCPODS-6776